src/Symfony/Component/Security/Core/Authorization/Voter/RoleVoter.php

```diff
@@ -37,6 +37,13 @@ public function __construct($prefix = 'ROLE_')
      */
     public function supportsAttribute($attribute)
     {
+        if (is_array($attribute)) {
+            return false;
+        }
+        if (is_object($attribute) && !method_exists($attribute, '__toString')) {
+            return false;
+        }
+
         return 0 === strpos($attribute, $this->prefix);
```

Comment:
Could be simply `(string) $attribtute` as well. No need for the above statements... it should crash for invalid values by design.